pkg/pillar/hypervisor/kvm.go

```diff
@@ -457,23 +457,138 @@ func (ctx kvmContext) Name() string {
 func (ctx kvmContext) Task(status *types.DomainStatus) types.Task {
 	if status.VirtualizationMode == types.NOHYPER {
 		return ctx.ctrdContext
-	} else {
-		return ctx
 	}
+	return ctx
 }
 
-// TBD: Have a better way to calculate this number.
-// For now it is based on some trial-and-error experiments.
-// Container limit is reduced to 100MiB.
-func minVMMOverhead(config types.DomainConfig) int64 {
+func estimatedVMMOverhead(domainName string, config types.DomainConfig, aa *types.AssignableAdapters) (int64, error) {
+	// Container limit is set to 100MiB.
+	// This is a temporary solution until we have a better way to predict
+	// the memory usage of the container.
 	if config.IsOCIContainer() {
-		return 100 << 20 // Mb in bytes
+		return 100 << 20, nil // Mb in bytes
```

Comment:
Didn't @rene say he sees ~300Mb for a container on arm64?

Reply:
@eriknordmark I'd like to handle ARM case in a separate PR and I need some time to do necessary calcualtions. I have prepared code for handling multi-function PCI device for NUC and it is based on refactored PCI code. For now @rene is using "memory.vmm.limit.MiB" property to override this limit for ARM

Reply:
@mikem-zed , in any case this PR must be tested and work for both. I would just change to 300 here and let it make it work on both until the ARM case can be addressed properly in a separated PR.